src/sagas/task.saga.js

```diff
@@ -0,0 +1,339 @@
+  // processId?
+  yield put(attachTaskMetadata(task, child.pid));
+
+  // When this application exits, we want to kill this process.
+  // Send it up to the main process.
+  yield call([ipcRenderer, ipcRenderer.send], 'addProcessId', child.pid);
+
+  const stdioChannel = createStdioChannel(child, {
+    stdout: emitter => data => {
+      // The 'eject' task prompts the user, to ask if they're sure.
+      // We can bypass this prompt, as our UI already has an alert that
+      // confirms this action.
+      // TODO: Eject deserves its own Redux action, to avoid cluttering up
+      // this generic "RUN_TASK" action.
+      // TODO: Is there a way to "future-proof" this, in case the CRA
+      // confirmation copy changes?
+      const isEjectPrompt = data
+        .toString()
+        .includes('Are you sure you want to eject? This action is permanent');
+
+      if (isEjectPrompt) {
+        sendCommandToProcess(child, 'y');
+      }
+
+      emitter({ channel: 'stdout', text: data.toString() });
+    },
+    stderr: emitter => data => {
+      emitter({ channel: 'stderr', text: data.toString() });
+    },
+    exit: emitter => code => {
+      const timestamp = new Date();
+
+      emitter({ channel: 'exit', timestamp, wasSuccessful: code === 0 });
+      emitter(END);
+    },
+  });
+
+  try {
+    while (true) {
+      const message = yield take(stdioChannel);
+
+      // eslint-disable-next-line default-case
+      switch (message.channel) {
+        case 'stdout':
+          yield put(receiveDataFromTaskExecution(task, message.text));
+          break;
+        case 'stderr':
+          yield put(receiveDataFromTaskExecution(task, message.text));
+          break;
+        case 'exit':
+          yield call(displayTaskComplete, task);
+          yield put(
+            completeTask(task, message.timestamp, message.wasSuccessful)
+          );
+          if (task.name === 'eject') {
+            yield put(loadDependencyInfoFromDisk(project.id, project.path));
+          }
+          break;
+      }
+    }
+  } finally {
+    /* child process has completed */
+  }
+}
+
+export function* taskAbort({ task }: { task: Task }): Saga<void> {
+  const { processId, name } = task;
+
+  yield call(killProcessId, processId);
+  yield call([ipcRenderer, ipcRenderer.send], 'removeProcessId', processId);
+
+  // Once the children are killed, we should dispatch a notification
+  // so that the terminal shows something about this update.
+  // My initial thought was that all tasks would have the same message,
+  // but given that we're treating `start` as its own special thing,
+  // I'm realizing that it should vary depending on the task type.
+  // TODO: Find a better place for this to live.
+  const abortMessage = isDevServerTask(name)
+    ? 'Server stopped'
+    : 'Task aborted';
+
+  yield put(receiveDataFromTaskExecution(task, chalk.bold.red(abortMessage)));
+}
+
+export function* displayTaskComplete(task: Task): Saga<void> {
```

Comment:
Looks like this method should also receive a `wasSuccessful` boolean, and then use that boolean to determine what the message should be ("Task completed" or "Task failed").

Tried this locally and indeed, I'm getting "Task complete" when the task fails:

<img width="670" alt="screen shot 2018-08-24 at 8 34 24 am" src="https://user-images.githubusercontent.com/6692932/44584918-87bc9f80-a778-11e8-9cec-d983f63ca9ce.png">